spec/lib/rmt/checksum_verifier_spec.rb

```diff
@@ -0,0 +1,156 @@
+require 'rails_helper'
+
+# rubocop:disable Metrics/LineLength
+
+RSpec.describe RMT::ChecksumVerifier do
+  let(:test_file_path) { fixture_file_path('checksum_verifier/file') }
+  let(:test_file_content) { fixture_file('checksum_verifier/file') }
+
+  it('has the content foobar') { expect(test_file_content).to eq('foobar') }
+
+  context 'hash function is unknown' do
+    it 'raises an exception' do
+      expect do
+        described_class.verify_checksum('CHUNKYBACON42', '0xDEADBEEF', test_file_path)
+      end.to raise_error(RMT::ChecksumVerifier::Exception, 'Unknown hash function CHUNKYBACON42')
+    end
+  end
+
+  context 'md5' do
+    context 'checksum is wrong' do
+      it 'raises an exception' do
+        expect do
+          described_class.verify_checksum('MD5', '0xDEADBEEF', test_file_path)
+        end.to raise_error(RMT::ChecksumVerifier::Exception, 'Checksum doesn\'t match')
+      end
+    end
+
+    context 'checksum is correct' do
+      it 'does not raises a exception' do
+        expect do
+          described_class.verify_checksum('MD5', '3858f62230ac3c915f300c664312c63f', test_file_path)
+        end.not_to raise_error
+      end
+
+      it 'handles strange formatting of MD5' do
+        expect do
+          described_class.verify_checksum('Md5', '3858f62230ac3c915f300c664312c63f', test_file_path)
```

Comment:
I would reduce the amount of examples by testing only `wEiRdLy` formatted/cased strings -- everything is upcased/downcased in the method anyway.
Also, hash value also contains case-sensitive characters.
Both aren't critical though.